arviz/utils.py

```diff
@@ -20,3 +21,20 @@ def _var_names(var_names):
 
     else:
         return var_names
+
+
+def conditional_jit(function=None, **kwargs):  # noqa: D202
+    """Use numba's jit decorator if numba is installed."""
+
+    def wrapper(function):
+        try:
+            numba = importlib.import_module("numba")
+            return numba.jit(**kwargs)(function)
+
+        except ImportError:
+            return function
+
+    if function:
+        return wrapper(function)
+    else:
```

Comment:
@aloctavodia This piece of code I'm confused on. In which situation would would  you need to use `conditional_jit` without a function?

Reply:
Not sure where this would be suitable and if it even works (probably), but isn't it kind of create a function that has all the settings done and then just apply it to some function (or maybe for many functions?)

Reply:
This allow to call it like `@conditional_jit` otherwise you will have to call it like `@conditional_jit()` or `@conditional_jit(some_keyword)`

Reply:
What's confusing to me is that in the tests It is being used as `@conditional_jit` but the else block is not hit according to coveralls

https://coveralls.io/builds/21049537/source?filename=arviz/utils.py#L40

Reply:
Can you print what is going on?

What about `function=None`?

Reply:
Sure, I will add more tests and some prints in the next ~1 hour. My gut feel though is that the else block is not needed if this will be used as a decorator because there will always be an object passed to the decorator.

I need to test with kwargs as well to ensure those are present.

Reply:
You will hit the "if-branch" when you use the decorator as `@conditional_jit` and the "else-branch when you use it as `@conditional_jit()` or  `@conditional_jit()`.

Reply:
Will update in ~5 hours and then we should be good